gnocchi/storage/_carbonara.py

```diff
@@ -117,9 +118,15 @@ def _store_metric_measures(metric, timestamp_key, aggregation,
                                granularity, data, offset=None, version=3):
         raise NotImplementedError
 
-    @staticmethod
-    def _list_split_keys_for_metric(metric, aggregation, granularity,
+    def _list_split_keys_for_metric(self, metric, aggregation, granularity,
                                     version=3):
+        return set(map(
+            functools.partial(carbonara.SplitKey, sampling=granularity),
+            utils.to_timestamps(self._list_split_keys(
```

Comment:
`_list_split_keys` returns time in seconds with nanoseconds precision.

> shouldn't we just pass _list_split_keys straight to SplitKey?

SplitKey wants a timestamps (= e.g `numpy.dateteim64`) as first arg. I can use `numpy.datetime64(int(x * 10e8, 'ns')` but it's what `to_timestamps` will do so I thought it's be simpler to just use it.